### PR TITLE
Reset LicenseString and Status property after request completed

### PR DESCRIPTION
**Pull request by raviteja-b**
Currently pldmd listens for LicenseString property change signal
and forwards license string to phyp.

This commit resets LicenseString and LicenseActivationStatus after request completed.
so that for further requests licenseString property change signal generated.

Increased  timeout to 20 sec to wait for response from pldmd
There are cases where phyp response arrives between 10-20 secs

This PR Fixes https://w3.rchland.ibm.com/projects/bestquest/?defect=SW544557
MUST FIX approved

Tested:
1. Upload new license key and check if its installed.
2. Upload same license key again and check if user got success
   instead of error

### CODE REVIEW

File: redfish-core/lib/license_service.hpp [hunk 1]
```diff
@@ -243,6 +286,11 @@ inline void requestRoutesLicenseEntryCollection(App& app)
                                     messages::internalError(asyncResp->res);
                                     return;
                                 }
+                                if (*status == "com.ibm.License.LicenseManager."
+                                               "Status.Pending")
+                                {
+                                    return;
```
Comment (sunharis) [inline redfish-core/lib/license_service.hpp:292]: So with this we are sending the empty response to the client, if the same license ? Which response code client is getting in this case

File: redfish-core/lib/license_service.hpp [hunk 2]
```diff
@@ -181,7 +224,7 @@ inline void requestRoutesLicenseEntryCollection(App& app)
             std::shared_ptr<boost::asio::steady_timer> timeout =
                 std::make_shared<boost::asio::steady_timer>(
                     crow::connections::systemBus->get_io_context());
-            timeout->expires_after(std::chrono::seconds(10));
+            timeout->expires_after(std::chrono::seconds(20));
             crow::connections::systemBus->async_method_call(
```
[the remark this answers is not in the document]
Comment (raviteja-b) [reply redfish-core/lib/license_service.hpp:228]: There are cases where phyp response arrives between 10-20 secs
Updated commit message

File: redfish-core/lib/license_service.hpp (hunk 1 above)
Comment (raviteja-b) [reply redfish-core/lib/license_service.hpp:292]: here we don't any response to client, this piece of code hit when bmcweb find a property change signal on LicenseActivationStatus.
this change is ignore D-bus signal if status value is Pending

**Comment by gtmills** [comment]
Must fix approved this but the actual fix I see is different:

    fix-complexity :
        level :
            Low

        description :

            Minor change to check if same license string on D-bus property already

**Comment by raviteja-b** [comment]
> Must fix approved this but the actual fix I see is different:
> 
> ```
> fix-complexity :
>     level :
>         Low
> 
>     description :
> 
>         Minor change to check if same license string on D-bus property already
> ```

yeah we discussed and decides to make these changes after MFB approval.

File: redfish-core/lib/license_service.hpp (hunk 1 above)
Comment (gtmills) [reply redfish-core/lib/license_service.hpp:292]: I think we need a comment here.. This also sounds pretty hacky

Comment (gtmills) [reply redfish-core/lib/license_service.hpp:292]: I think we need a comment here.. This also sounds pretty hacky

Comment (gtmills) [reply redfish-core/lib/license_service.hpp:296]: Can the comment say “why” we are doing this

**Review by gtmills — APPROVED** [review]

**Pull request merged**